includes/class-wc-payments.php

```diff
@@ -244,6 +244,26 @@ public static function init() {
 		include_once __DIR__ . '/exceptions/class-base-exception.php';
 		include_once __DIR__ . '/exceptions/class-api-exception.php';
 		include_once __DIR__ . '/exceptions/class-connection-exception.php';
+		include_once __DIR__ . '/core/class-mode.php';
+
+		// Include core exceptions.
+		include_once __DIR__ . '/core/exceptions/class-invalid-request-parameter-exception.php';
+		include_once __DIR__ . '/core/exceptions/class-immutable-parameter-exception.php';
+		include_once __DIR__ . '/core/exceptions/class-extend-request-exception.php';
+		include_once __DIR__ . '/core/exceptions/class-server-response-exception.php';
+		include_once __DIR__ . '/core/exceptions/class-server-request-exception.php';
```

Comment:
Should we maybe namespace the Exceptions with `Server` same as for `core/server` classes? 

This namespace + class name combination 
```php
/core/exceptions/class-server-response-exception.php
```

explain the scope very well, however this 
```php
/core/exceptions/class-extend-request-exception.php
```

isn't. The name indicates that it can be used globally but implementation is meant only for specific `Server/` namespace use.

Reply:
I agree with you on this topic. Will refactor.

Reply:
This is done.  all request exceptions are moved to its own directory.